src/cuda-headers/gpu_common.h

```diff
@@ -21,4 +33,61 @@ inline void cuda_assert(cudaError_t err, const char *file, int line)
     }
 }
 
+/************************************
```

Comment:
I would split this into two files something like: gpu_cuda_common.h and gpu_opencl_common.h. The OpenCL and cuda will not share this logic probably.